packages/arb-node-core/cmd/arb-relay/arb-relay.go

```diff
@@ -136,31 +139,41 @@ func (ar *ArbRelay) Start(ctx context.Context) (chan bool, error) {
 	}
 
 	// connect returns
-	var messages chan broadcaster.BroadcastFeedMessage
-	for {
-		messages, err = ar.broadcastClient.Connect(ctx)
-		if err == nil {
-			break
-		}
-		logger.Warn().Err(err).
-			Msg("failed connect to sequencer broadcast, waiting and retrying")
-
-		select {
-		case <-ctx.Done():
-			return nil, errors.New("ctx cancelled broadcast client connect")
-		case <-time.After(5 * time.Second):
+	messages := make(chan broadcaster.BroadcastFeedMessage)
+	for _, client := range ar.broadcastClients {
+		for {
+			err = client.ConnectWithChannel(ctx, messages)
+			if err == nil {
+				break
+			}
+			logger.Warn().Err(err).
+				Msg("failed connect to sequencer broadcast, waiting and retrying")
+			select {
+			case <-ctx.Done():
+				return nil, errors.New("ctx cancelled broadcast client connect")
+			case <-time.After(5 * time.Second):
+			}
 		}
 	}
 
+	recentFeedItems := make(map[common.Hash]time.Time)
 	go func() {
 		defer func() {
 			done <- true
 		}()
+		recentFeedItemsCleanup := time.NewTicker(RECENT_FEED_ITEM_TTL)
 		for {
 			select {
 			case <-ctx.Done():
+				recentFeedItemsCleanup.Stop()
 				return
 			case msg := <-messages:
+				newAcc := msg.FeedItem.BatchItem.Accumulator
+				if recentFeedItems[newAcc] == (time.Time{}) {
+					recentFeedItems[newAcc] = time.Now()
```

Comment:
Then we'd also want the error handling inside the if statement, and in general, this pattern means that the happy case code sits less indented and not in a special condition. I prefer this because the special case is skipping the item, not processing it.

Reply:
Makes total sense, how about this instead?
```
if recentFeedItems[newAcc] != (time.Time{}) {
    continue
}
```

That way you can still get rid of the else statement, one line less indented